lib/elixir/lib/kernel/utils.ex

```diff
@@ -6,8 +6,11 @@ defmodule Kernel.Utils do
   @doc """
   Callback for destructure.
   """
-  def destructure(list, count) when is_list(list), do: destructure_list(list, count)
-  def destructure(nil, count), do: destructure_nil(count)
+  @spec destructure(list | nil, non_neg_integer) :: list
```

Comment:
No need to spec this function. This is a callback for internal implementation details.